### DISCUSSION TITLE
viper cannot be used in init function, Right?

### DISCUSSION THREAD

**Discussion by pplmx** [spf13/viper#1279] [Q&A] [answered]
# Here is my code
```go
var (
	Rdb *redis.Client
	Ctx context.Context
)

func init() {
	fmt.Println("Start to init conn package >>>>>>>>> ")
	redisUrl := viper.Get("datasource.redis.url").(string)
	fmt.Println("Failed to parse configuration file")

	Rdb = db.RedisClient(redisUrl)
	Ctx = context.Background()
}

```

## Error message
It doesn't print "Failed to parse ...."
```text
Start to init conn package >>>>>>>>>
panic: interface conversion: interface {} is nil, not string

goroutine 1 [running]:

```

**Comment by sagikazarmark** [comment, marked as answer]
Well, there is nothing that keeps you from using it in an init function, but if there is nothing available under a certain key, it will return nil.

Viper doesn't load files automatically. See the readme for examples.

Technically, the offending line is this: `redisUrl := viper.Get("datasource.redis.url").(string)`

You could change it to make the error go away, but it would still not return anything: `redisUrl, ok := viper.Get("datasource.redis.url").(string)`

**Reply by pplmx** [reply]
Thanks 😄
